.github/workflows/scheduled-ansible-pull.yml

```diff
@@ -0,0 +1,58 @@
+name: Check if the collection is up to date with the ansible role
+
+on:
+  - workflow_dispatch
+  - schedule:
+    - cron: '0 4 * * *'
+
+jobs:
+  check:
+    runs-on: ubuntu-latest
+    outputs:
+      check-output: ${{ steps.check-diff.outputs.sync-needed }}
+    steps:
+      - name: Check diff between roles/agent and ansible-role main branch
+        id: check-diff
+        run: |
+          mkdir /tmp/ansible-role
+          git clone git@github.com:DataDog/ansible-datadog.git /tmp/ansible-role
+          ERROR_CODE=0
+          diff -x "\.*" -x CHANGELOG.md -x CONTRIBUTING.md -x LICENSE -x ci_test -x manual_tests -x "requirements*" -pur roles/agent/ /tmp/ansible-role || ERROR_CODE=$?
+          if [ $ERROR_CODE -ne 0 ]; then
+            echo "sync-needed=true" >> $GITHUB_OUTPUT
+          fi
```

Comment:
```suggestion
          diff -x "\.*" -x CHANGELOG.md -x CONTRIBUTING.md -x LICENSE -x ci_test -x manual_tests -x "requirements*" -pur roles/agent/ /tmp/ansible-role || "sync-needed=true" >> $GITHUB_OUTPUT
```

Reply:
I'm not entirely sure if this will work as intended with the || and the file redirection, but I don't think `ERROR_CODE` is needed for this purpose

Reply:
Agreed, tbh since it's not yet ready for review 😆 I just put it out there to test it. I was planning to remove it

Reply:
Oh my bad! I received a notification so I took a look, but please free to ignore for the time being and I'll review again once ready :)

Reply:
No worries still helpful to have feedback so I have less re-reading/refacto to do 💯